mirrord/agent/src/outgoing/udp.rs

```diff
@@ -265,3 +265,34 @@ impl UdpOutgoingApi {
             .ok_or(AgentError::ReceiverClosed)
     }
 }
+
+pub(crate) struct SendRecvManager {
+    open_sockets: HashMap<SocketAddr, UdpSocket>,
+}
+
+impl SendRecvManager {
+    pub(crate) fn new() -> Self {
+        Self {
+            open_sockets: HashMap::new(),
+        }
+    }
+
+    pub(crate) async fn handle_message(
+        &self,
+        request: SendRecvRequest,
+    ) -> Result<Option<SendRecvResponse>> {
+        match request {
+            SendRecvRequest::SendMsg(message) => {
+                let socket = UdpSocket::bind(message.addr.clone()).await?;
```

Comment:
the address in the message is of destination, we can't bind it (binding is for local)

Reply:
right it should be localhost

Reply:
you don't need to bind..

Reply:
```rs
This function will create a new UDP socket and attempt to bind it to the addr provided.

Binding with a port number of 0 will request that the OS assigns a port to this listener. The port allocated can be queried via the local_addr method.
```